### PR TITLE
Process data

### PR DESCRIPTION
**Pull request by metavore**
- Now using aggregator instead of ring buffer. 
- Larger window size. 
- Discovered and accounted for fixed frequency behavior of synthetic board in tests.
- Code is cleaner, leaner, and more readable.

### CODE REVIEW

File: neuros/main.py [hunk 1]
```diff
@@ -58,34 +68,51 @@ def board_stream(board_id: int = BoardIds.SYNTHETIC_BOARD,
...
 
-                filled_samples += new_data.shape[1]
+                    # Remove processed data, keeping overlap if needed
+                    aggregator = aggregator[:, needed:]
 
-                if filled_samples >= window_samples:
-                    yield ring_buffer.copy()
+                    yield window
 
             except Exception as e:
```
Comment (EricHennigan) [inline neuros/main.py:117]: improve readability by moving the exception catch closer to the line that throws

File: neuros/main.py [hunk 2]
```diff
@@ -58,34 +68,51 @@ def board_stream(board_id: int = BoardIds.SYNTHETIC_BOARD,
...
 
-                # Extract just the EEG channels from the new data
+                # Extract EEG channels and add to aggregator
                 eeg_data = new_data[eeg_channels]
+                aggregator = np.concatenate([aggregator, eeg_data], axis=1)
 
-                ring_buffer[:, :-new_samples] = ring_buffer[:, new_samples:]
-                ring_buffer[:, -new_samples:] = eeg_data
+                # Generate windows while we have enough data
+                while aggregator.shape[1] >= window_samples:
```
Comment (EricHennigan) [inline neuros/main.py:108]: Can you explain why there is a second while loop? I naively expected the outer loop to be enough, so I must be missing something

File: tests/test_streams.py [hunk 3]
```diff
@@ -122,5 +122,20 @@ def test_eeg_channel_count():
         assert window.shape[0] == 16  # Total number of eeg channels for synthetic board
 
 
+def test_ring_buffer_shift():
```
Comment (EricHennigan) [inline tests/test_streams.py:125]: Is this code needed? I see you've removed the ring buffer computation from main.py

File: tests/test_window_streams.py [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+    WindowConfig(window_ms=100.0, overlap_ms=60.0)  # Another known good set
+
+    # Invalid configurations
+    with pytest.raises(ValueError):
+        WindowConfig(window_ms=100.0, overlap_ms=100.0)  # Equal window and overlap
+    with pytest.raises(ValueError):
+        WindowConfig(window_ms=100.0, overlap_ms=150.0)  # Overlap > window
+
+
+def test_basic_window_shape():
```
Comment (EricHennigan) [inline tests/test_window_streams.py:19]: Can you rephrase some of these test to make it more clear what is the scenario and expectation?
The Given...When...Then paradigm in the example at https://cucumber.io/ is a pretty solid pattern

File: neuros/processing.py
Comment (EricHennigan) [inline neuros/processing.py:1]: I don't see this being used in main.py
Is it early?

**Review by EricHennigan — COMMENTED** [review]
The commit line is not informative. It doesn't say what got fixed

File: tests/test_window_streams.py [hunk 5]
```diff
@@ -129,8 +125,11 @@ def test_error_handling():
         # 1. Return empty/zero data
         # 2. Skip the error and continue on next valid data
         # 3. Or maintain last valid state
-        window2 = next(generator)
-        assert window2 is not None  # Shouldn't crash
+        try:
```
Comment (EricHennigan) [inline tests/test_window_streams.py:128]: This test doesn't make much sense. The assertions are mutually exclusive in execution, and I don't know which one is supposed to run

File: neuros/main.py (hunk 1 above)
Comment (metavore) [reply neuros/main.py:117]: Old version, this no longer exists.

File: neuros/main.py (hunk 2 above)
Comment (metavore) [reply neuros/main.py:108]: Old version, this no longer exists.

File: tests/test_streams.py (hunk 3 above)
Comment (metavore) [reply tests/test_streams.py:125]: Old version, this no longer exists.

File: tests/test_window_streams.py (hunk 4 above)
Comment (metavore) [reply tests/test_window_streams.py:19]: Will do, thanks for the cucumber.

File: neuros/processing.py
Comment (metavore) [reply neuros/processing.py:1]: Yes, I took it out as it had a non-fatal error noising up my debugging. It should be in a different branch.

File: tests/test_window_streams.py (hunk 5 above)
Comment (metavore) [reply tests/test_window_streams.py:128]: That's the 5-minutes-before-party code XD Will clean up the tests and make them all clear, informative and relevant.

**Comment by metavore** [comment]
I may have mixed up the branches a bit. process-data was supposed to come after data-acquisition, keeping the goals of each clear. I'll straighten all of that up shortly, and PR both with the correct stuff in the correct sequence,

**Comment by metavore** [comment]
Yep. All of the current work is supposed to come through the other branch. This one is for the next part, so I'll PR this one only after the other one gets pulled.
